android/app/src/main/java/app/organicmaps/location/TrackRecordingService.java

```diff
@@ -0,0 +1,249 @@
+        .setCategory(NotificationCompat.CATEGORY_SERVICE)
+        .setPriority(NotificationManager.IMPORTANCE_DEFAULT)
+        .setVisibility(NotificationCompat.VISIBILITY_SECRET)
+        .setOngoing(true)
+        .setShowWhen(true)
+        .setOnlyAlertOnce(true)
+        .setSmallIcon(R.drawable.ic_splash)
+        .setContentTitle(context.getString(R.string.recent_track_recorder_running))
+        .setContentText(context.getString(R.string.recent_track_rec_notif_desc))
+        .setContentIntent(getPendingIntent(context))
+        .setColor(ContextCompat.getColor(context, R.color.notification));
+
+    return mNotificationBuilder;
+  }
+
+  public static void stopService(@NonNull Context context)
+  {
+    Logger.i(TAG);
+    context.stopService(new Intent(context, TrackRecordingService.class));
+  }
+
+  @Override
+  public void onDestroy()
+  {
+    mNotificationBuilder = null;
+    mWarningBuilder = null;
+    if (TrackRecorder.nativeIsEnabled())
+      TrackRecorder.nativeSetEnabled(false);
+    mHandler.removeCallbacks(mLocationTimeoutRunnable);
+    LocationHelper.from(this).removeListener(this);
+    // The notification is cancelled automatically by the system.
+  }
+
+  @Override
+  public int onStartCommand(@NonNull Intent intent, int flags, int startId)
+  {
+    if (!MwmApplication.from(this).arePlatformAndCoreInitialized())
+    {
+      Logger.w(TAG, "Application is not initialized");
+      stopSelf();
+      return START_NOT_STICKY; // The service will be stopped by stopSelf().
+    }
+
+    if (!LocationUtils.checkFineLocationPermission(this))
+    {
+      // In a hypothetical scenario, the user could revoke location permissions after the app's process crashed,
+      // but before the service with START_STICKY was restarted by the system.
+      Logger.w(TAG, "Permission ACCESS_FINE_LOCATION is not granted, skipping TrackRecordingService");
+      stopSelf();
+      return START_NOT_STICKY; // The service will be stopped by stopSelf().
+    }
+
+    if (!TrackRecorder.nativeIsEnabled())
+    {
+      Logger.i(TAG, "Service can't be started because Track Recorder is turned off in settings");
+      stopSelf();
+      return START_NOT_STICKY;
+    }
+
+    Logger.i(TAG, "Starting foreground service");
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S)
+    {
+      try
+      {
+        startForeground(TrackRecordingService.TRACK_REC_NOTIFICATION_ID, getNotificationBuilder(this).build());
+      } catch (ForegroundServiceStartNotAllowedException e)
+      {
+        Logger.e(TAG, "Oops! ForegroundService is not allowed", e);
+      }
+    }
+    else
+    {
+      startForeground(TrackRecordingService.TRACK_REC_NOTIFICATION_ID, getNotificationBuilder(this).build());
+    }
+
+    final LocationHelper locationHelper = LocationHelper.from(this);
+
+    if (mHandler == null) mHandler = new Handler();
+    if (mLocationTimeoutRunnable == null) mLocationTimeoutRunnable = this::onLocationUpdateTimeout;
```

Comment:
Don't use one-line ifs, they're harder to debug.

Reply:
What is the preferred way then?

Reply:
```
if (x)
  do = y;
```